src/validator.js

```diff
@@ -0,0 +1,64 @@
+import * as fs from 'fs';
+import Xpi from  'xpi';
+
+import promisify from 'es6-promisify';
+import chalk from 'chalk';
+
+export var lstat = promisify(fs.lstat);
+
+
+export default class Validator {
+
+  constructor(config) {
+    this.config = config;
+    this.packagePath = config._[0];
+    this.xpi;
+    this.chalk = new chalk.constructor(
+      {enabled: !this.config.boring});
+  }
+
+  handleError(err) {
+    if (this.config.stack === true) {
+      console.error(err.stack);
+    } else {
+      console.error(this.chalk.red(err.message || err));
+    }
+  }
+
+  checkFileExists(filepath, _lstat=lstat) {
+    var invalidMsg = new Error(
+      'Path given is not a file or does not exist: "' + filepath + '"');
```

Comment:
And actually, reading the output I find this sentence hard to parse. This might read better:

```js
var invalidMessage = new Error(
  `Path "${filepath}" is not a file or does not exist.`);
```

This way you can see the path right away and the sentence is easier to mentally parse. I find that worth the trade-off of not seeing the path clearly at the end of the error (it's in quotes anyway, so it stands out).